finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/Message.scala

```diff
@@ -122,6 +128,15 @@ object Request {
     }
     override def authority = headers.get(Headers.Authority).getOrElse("")
     override def path = headers.get(Headers.Path).getOrElse("/")
+    override def transform(scheme: String, method: Method, authority: String, path: String) = copy(
+      Headers(
+        Headers.Scheme -> scheme,
+        Headers.Method -> method.toString,
+        Headers.Authority -> authority,
+        Headers.Path -> path
+      ),
+      stream
+    )
```

Comment:
good catch, dough I didn't expect an immutable type to have a mutable field

how do you propose to fix that?